Lib/test/test__opcode.py

```diff
@@ -7,23 +7,80 @@
 from _opcode import stack_effect
 
 
-class OpcodeTests(unittest.TestCase):
-
-    def check_bool_function_result(self, func, ops, expected):
-        for op in ops:
-            if isinstance(op, str):
-                op = dis.opmap[op]
-            with self.subTest(opcode=op, func=func):
-                self.assertIsInstance(func(op), bool)
-                self.assertEqual(func(op), expected)
-
+EXPECTED_OPLISTS = {
```

Comment:
Does this mean that each time we add or remove an opcode we need to update this test? That feels tedious. (Even though it looks like it was always done like this and you're just refactoring that code a bit.)

Reply:
In the past the list was built in opcode.py so we would need to make sure we added i there. Now the lists are derived automatically from bytecodes.c via heuristics, and you will need to update the test (which you are less likely to be able to do incorrectly). I'm not sure it's safe to not have these full tests.

Reply:
Hm. So the point of the heuristics is that we don't have to manually maintain it, but the tests are manually maintained to check that the heuristics are still working. If they are, and if an instruction is added (or deleted) that changes one of these lists, we'll get a bogus test failure, and hopefully the contributor who gets to debug the test failure understands they have to fix the test. I think this warrants a comment right there where we expect the test to fail (I think on line 102, `self.assertEqual(res, op in expected)`).

Reply:
The manually maintained heuristics were not tested and had some errors. We could remove the tests and trust people to examine the diff in the metadata file when they make changes. If we do that we should remove it from the autogenerated list so that it's not collapsed in a PR diff.